### PR TITLE
Remove `create_new_model_version` arg of `ModelConfig`

### PR DESCRIPTION
**Pull request by avishniakov**
## Describe changes
I simplified the `ModelConfig` interface to avoid explicitly passing around `create_new_model_version`.
New behavior expects that the user keeps `version==None` if the new unnamed version is needed, otherwise model version will be retrieved/created based on the `version` argument.
Also new `ModelStages.LATEST` was introduced in order to give the ability to fetch the LATEST model version.

**[BREAKING CHANGE]** Old configurations using `create_new_model_version` might lead to unpredicted results, but the change needed is simple:
```python
ModelConfig(name="foo",create_new_model_version=True) -> ModelConfig(name="foo")
ModelConfig(name="foo", version="bar", create_new_model_version=True) -> ModelConfig(name="foo", version="bar")
ModelConfig(name="foo", version="bar", create_new_model_version=False) -> ModelConfig(name="foo", version="bar")
ModelConfig(name="foo", version=None, create_new_model_version=False) -> ModelConfig(name="foo", version=ModelStages.LATEST)
```

**[BEHAVIOR CHANGE]** If a signature like `ModelConfig(name="foo", version="bar", create_new_model_version=True)` was used, it prevented 2 parallel pipelines from running pointing to same model version, this is not the case anymore for named versions.

Tests updates mostly relate to signature updates, but also some tests were removed as no longer valid ones.

## Pre-requisites
Please ensure you have done the following:
- [x] I have read the **CONTRIBUTING.md** document.
- [ ] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] If I have added an integration, I have updated the [integrations](https://docs.zenml.io/stacks-and-components/component-guide) table and the [corresponding website section](https://zenml.io/integrations).
- [x] I have added tests to cover my changes.
- [x] I have based my new branch on `develop` and the open PR is targeting `develop`. If your branch wasn't based on develop read [Contribution guide on rebasing branch to develop](https://github.com/zenml-io/zenml/blob/main/CONTRIBUTING.md#-pull-requests-rebase-your-branch-on-develop).

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

### CODE REVIEW

File: src/zenml/model/model_config.py [hunk 1]
```diff
@@ -151,18 +118,25 @@ def _validate_config_in_runtime(self) -> None:
...
-            if self.create_new_model_version:
+            if self.version is None or self.version == RUNNING_MODEL_VERSION:
+                self.version = RUNNING_MODEL_VERSION
                 for run_name, run in model_version.pipeline_runs.items():
                     if run.status == ExecutionStatus.RUNNING:
                         raise RuntimeError(
-                            f"New model version was requested, but pipeline run `{run_name}` "
-                            f"is still running with version `{model_version.name}`."
+                            f"New unnamed model version was requested, "
+                            f"but pipeline run `{run_name}` have not finished yet. "
```
Comment (fa9r) [inline src/zenml/model/model_config.py:127]:
```suggestion
                            f"but pipeline run `{run_name}` has not finished yet. "
```

File: src/zenml/model/model_config.py [hunk 2]
```diff
@@ -151,18 +118,25 @@ def _validate_config_in_runtime(self) -> None:
...
+                            "This run also operates with unnamed model version - "
+                            "new run will be stopped to prevent unexpected behavior."
```
Comment (fa9r) [inline src/zenml/model/model_config.py:129]: I don't fully get the error message - would be better if you explain what the user is supposed to do now? Rename his version? Or wait for this other run to complete?

File: src/zenml/model/model_config.py [hunk 3]
```diff
@@ -307,3 +260,22 @@ def _merge(self, model_config: "ModelConfig") -> None:
         self.delete_new_version_on_failure &= (
             model_config.delete_new_version_on_failure
         )
+
+    def __hash__(self) -> int:
```
Comment (fa9r) [inline src/zenml/model/model_config.py:264]: what's this used for?

**Review by fa9r — CHANGES_REQUESTED** [review]
The main change looks good but I'm not a fan of the changes to the client, models, and zenstore methods. For these, fetching the latest version by default made sense and was more convenient to use with the version arg being optional. I'd suggest reverting these changes as indicated in the comments below, otherwise LGTM.

File: src/zenml/model/model_config.py (hunk 3 above)
Comment (avishniakov) [reply src/zenml/model/model_config.py:264]: [Here](https://github.com/zenml-io/zenml/blob/07b93568230a0eb1ca6ba6a75d67448b729bc869/src/zenml/new/pipelines/pipeline.py#L851)

File: src/zenml/model/model_config.py (hunk 2 above)
Comment (avishniakov) [reply src/zenml/model/model_config.py:129]: reworked

**Comment by avishniakov** [comment]
@fa9r , tests passed, if you don't see any more changes needed here, I would love to merge this to proceed with renaming ticket.

**Review by fa9r — APPROVED** [review]
LGTM

**Pull request merged**